movingpandas/trajectory_cleaner.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+
+import pandas as pd
+from copy import copy
+
+from .trajectory import Trajectory
+from .trajectory_collection import TrajectoryCollection
+
+
+class TrajectoryCleaner:
+    """
+    Cleaner base class
+    """
+
+    def __init__(self, traj):
+        """
+        Create TrajectoryCleaner
+
+        Parameters
+        ----------
+        traj : Trajectory or TrajectoryCollection
+        """
+        self.traj = traj
+
+    def clean(self, columns):
+        """
+        Clean the input Trajectory/TrajectoryCollection.
+
+        Parameters
+        ----------
+        columns: dictionary
+            Information regarding the columns that will be used to clean the trajectory
+            and the accompanying thresholds. ex. - {'speed': 30, 'heading': 360} etc.
+
+        Returns
+        -------
+        Trajectory/TrajectoryCollection
+            Cleaned Trajectory or TrajectoryCollection
+        """
+        if isinstance(self.traj, Trajectory):
+            return self._clean_traj(self.traj, columns)
+        elif isinstance(self.traj, TrajectoryCollection):
+            return self._clean_traj_collection(columns)
+        else:
+            raise TypeError
+
+    def _clean_traj_collection(self, columns):
+        cleaned = []
+        for traj in self.traj:
+            cleaned.append(self._clean_traj(traj, columns))
+        result = copy(self.traj)
+        result.trajectories = cleaned
+        return result
+
+    def _clean_traj(self, traj, columns):
+        return traj
+
+
+class OutlierCleaner(TrajectoryCleaner):
+    """
+    Outlier (interquantile range - iqr) based cleaner.
+
+    columns : dictionary
+        Key - value pairs of columns and alpha (iqr multiplier).
+
+        Note: Setting alpha=3 is widely used.
+
+    Examples
+    --------
+
+    >>> mpd.OutlierCleaner(traj).clean({'speed': 3})
+    """
+
+    def _clean_traj(self, traj, columns):
+        df = traj.df.copy()
+        ixs = []
+
+        for column, alpha in columns.items():
+            ix = self._calc_outliers(df[column], alpha)
```

Comment:
Did you check what error message we get if the column is not numeric? Is it a good descriptive error message or do we need a custom one?

Reply:
The error message is ugly indeed. I wrote a nice one that specifies that the column with name "something" of type "somthing_non_numeric" is not numeric (used pandas.api.types.is_numeric).

Looks like this:
<img width="762" alt="image" src="https://user-images.githubusercontent.com/15364873/148452787-9de79ff4-cb3a-468d-9a96-3b96a8b12233.png">